File: ios/Runner/Pusher/Channels/Call Channel/CallChannelEvents.swift
//
//  CallChannelEvents.swift
//  Runner
//
//  Created by Hashim Khan on 24/01/2024.
//


enum CallChannelEvents {
    
    case incommingCall
    case incommingCallDeclined
    case callAccepted
    case callDeclined
    case callEnded
    case userBueasy
    case callRinging
    case noAnswer
    
    
    public func getName() -> String {
        switch self {
        case .incommingCall: return "incomming-call"
        case .incommingCallDeclined: return "incomming-call-declined"
        case .callAccepted: return "call-accepted"
        case .callDeclined: return "call-declined"
        case .callEnded: return "call-ended"
        case .userBueasy: return "user-bueasy"
        case .callRinging: return "call-ringing"
        case .noAnswer: return "no-answer"
        }
    }
}

File: android/app/src/main/kotlin/com/transport_system/ts_driver/pusher/channels/call_channel/enums/CallChannelEvents.kt
package com.transport_system.ts_driver.pusher.channels.call_channel.enums

enum class CallChannelEvents {
    INCOMING_CALL,
    INCOMING_CALL_DECLINED,
    CALL_ACCEPTED,
    CALL_DECLINED,
    CALL_ENDED,
    USER_BUSY,
    CALL_RINGING,
    NO_ANSWER;

    fun getName(): String {
        return when (this) {
            INCOMING_CALL -> "incomming-call"
            INCOMING_CALL_DECLINED -> "incomming-call-declined"
            CALL_ACCEPTED -> "call-accepted"
            CALL_DECLINED -> "call-declined"
            CALL_ENDED -> "call-ended"
            USER_BUSY -> "user-bueasy"
            CALL_RINGING -> "call-ringing"
            NO_ANSWER -> "no-answer"
        }
    }
}
